File: Modules/DiagnosticsLogger.vb
''' <summary>
''' Centralized diagnostics logger for FBEditor.
''' Replaces silent Catch blocks with logged catches so errors are diagnosable.
''' Output goes to System.Diagnostics.Debug (visible in VS Output window).
''' </summary>
Public Module DiagnosticsLogger

    Public Sub LogWarning(source As String, message As String)
        System.Diagnostics.Debug.WriteLine($"[FBEditor][{source}] WARNING: {message}")
    End Sub

    Public Sub LogError(source As String, message As String, Optional ex As Exception = Nothing)
        Dim msg = $"[FBEditor][{source}] ERROR: {message}"
        If ex IsNot Nothing Then msg &= $" | {ex.GetType().Name}: {ex.Message}"
        System.Diagnostics.Debug.WriteLine(msg)
    End Sub

    Public Sub LogInfo(source As String, message As String)
        System.Diagnostics.Debug.WriteLine($"[FBEditor][{source}] {message}")
    End Sub

End Module

File: FBEdior-Linux-x64/FBEditor.Core/DiagnosticsLogger.cs
namespace FBEditor.Core;

/// <summary>
/// Centralized diagnostics logger for FBEditor.
/// Replaces silent catch blocks with logged catches so errors are diagnosable.
/// Output goes to System.Diagnostics.Debug.
/// Ported from Modules/DiagnosticsLogger.vb (VB Module -> C# static class).
/// </summary>
public static class DiagnosticsLogger
{
    public static void LogWarning(string source, string message) =>
        System.Diagnostics.Debug.WriteLine($"[FBEditor][{source}] WARNING: {message}");

    public static void LogError(string source, string message, Exception? ex = null)
    {
        var msg = $"[FBEditor][{source}] ERROR: {message}";
        if (ex != null) msg += $" | {ex.GetType().Name}: {ex.Message}";
        System.Diagnostics.Debug.WriteLine(msg);
    }

    public static void LogInfo(string source, string message) =>
        System.Diagnostics.Debug.WriteLine($"[FBEditor][{source}] {message}");
}
